### DISCUSSION TITLE
Community Building

### DISCUSSION THREAD

**Discussion by bhazelton** [cssi-pi-community/2022-meeting#110] [Special topics breakouts (Tuesday afternoon)]
# Discussion items from the Community Building breakout

**Comment by erjank** [comment]
Importance of responsivity to community questions, accessibility of software.

**Comment by erjank** [comment]
Codes of conduct: Does your community have one? Is it enforced?

**Comment by tjhacker** [comment]
Software infrastructure for communication is important for community building.

**Comment by bhazelton** [comment]
What are best practices and approaches to building welcoming, supportive communities around open source code?

**Reply by erjank** [reply]
A hypothesis from me: Being intentional and forward about it with a code of conduct. Kicking off events with "This is a welcoming space where harassment in any form is not tolerated" really goes a long way, and teaching in teams so that you're never alone in handling any conduct issues are two elements I'd raise.

**Reply by bhazelton** [reply]
from discussion:
- some contributors become very good at answering questions from new users. They are particularly helpful creating  a welcoming community.

**Reply by bhazelton** [reply]
Ask new people:

What do you expect to get out of this? 

Be thoughtful about what is in it for them and what can they give back to the community.

**Comment by erjank** [comment]
In-person workshop events as a catalyst for community-building. Discuss!

**Reply by tjhacker** [reply]
Support for community building for individuals involved should be more sustainable and deliberate instead of asking individuals to volunteer or simply add efforts to their existing workload.

**Reply by erjank** [reply]
Weeklong, both in-person and virtual models of success at our table.

**Reply by tjhacker** [reply]
Helpful book for consideration:
Krile, James F., Gordon J. Curphy, and Duane Richard Lund. The community leadership handbook: Framing ideas, building relationships, and mobilizing resources. Fieldstone Alliance, 2006.

from this book, page xiv near the front provides a helpful definition of the meaning of community leadership as well as the characteristics of a "healthy community" that seeks to work to meet the myriad needs of the members of the community and that this type of  community seeks to "work together for the common good"

**Reply by erjank** [reply]
Ooh, thanks!

**Reply by bhazelton** [reply]
More from the book referenced above:

Start by identifying community assets and community needs.

**Reply by tjhacker** [reply]
it is important to build, sustain, and include new people.

**Reply by bhazelton** [reply]
Working to make new connections in person -- don't let people just talk to people they know. Good example was table assignments at this conference.

**Reply by tjhacker** [reply]
it also also important to consider how to include people and interact beyond electronic means such as email.

**Comment by namurphy** [comment]
- Expression of surprise that not everyone here is in computer science
 - Community of communities are helpful: US-RSE, [Carpentries](https://carpentries.org/) (Software/Data Carpentry), Science Gateways
   - Have a calendar of events on website with other communities
   - Need a place to come together
 - How to build a community from scratch?  Both user community & developer community.
   - Carpentries style model? In this case, might be too small for a Carpentries style training.
   - Maybe have people take the training, and then motivate them to bring the trainings to their own groups.
   - Common reference materials
   - Instructor trainings by the Carpentries
   - Carpentries materials were adapted for a subfield
   - For Einstein toolkit, organize a workshop in U.S. and a workshop in Europe each year
   - Example is doing a hack week (e.g., Plasma Hack Week, Astro Hack Week)

**Reply by erjank** [reply]
I've found "forking" Carpentries-style training to be very effective and glad to chat more with anyone about that!

**Reply by bhazelton** [reply]
> * Expression of surprise that not everyone here is in computer science

Curious about the context of this?

**Reply by namurphy** [reply]
> > * Expression of surprise that not everyone here is in computer science
> 
> Curious about the context of this?

One of the participants is a first-time CSSI PI, and coming into this meeting, had thought that most of the people who would be at this meeting would be from computer science departments.

**Reply by erjank** [reply]
Think the CS context was for  @namurphy 's comment above, but I'm curious too!

**Comment by bhazelton** [comment]
## Specific questions about the CSSI PI community:
- Should we have more structure beyond an annual meeting (like this one)?
- Is it sustainable to rely on PIs to organize and run this meeting and any other community structure we desire

**Reply by erjank** [reply]
Oooh, really good points. Something like a quarterly opportunity to check back in on discussions like these would be great, and support for organizing those sorts of activities would be a nice sustainability effort.

**Reply by klehnert55** [reply]
I truly think that CSSI could be the start of a cross-disciplinary CI community that maintains dialog and exchange. It would need a way to support ongoing activities. I mentioned the EarthCube Office as an example for an activity that supported community activities.

**Reply by bhazelton** [reply]
from our discussion:
This should probably include people who do similar work but are not funded through CSSI. In fact, people who are not funded through CSSI probably need the community more than we do!

**Reply by klehnert55** [reply]
Who is this community? Only CSSI or a broader group of CI PIs?

**Comment by namurphy** [comment]
- Can put code on [nanohub](https://nanohub.org/)
 - For Citizen Science, [Zooniverse](https://www.zooniverse.org/)
   - Like, counting penguins in a picture or classifying galaxies which are hard to automate

**Reply by klehnert55** [reply]
Look at Community Leadership Handbook!

**Comment by erjank** [comment]
Discussion of not distinguishing users from developers email lists for managing projects

**Reply by erjank** [reply]
That is, having one mailing list instead of two.

**Reply by bhazelton** [reply]
I think identifying people who contribute issues/bug reports as "contributors" to software is also important and may be related to the sentiment behind this.

**Comment by namurphy** [comment]
- What are ~~top~~ cyberinfrastructure publications that are most appropriate for CSSI projects to publish in?
   - Future Generation of Computer Systems
   - Computational Experience & Practice
   - JOSS
   - For HPC, supercomputing ones

**Reply by erjank** [reply]
https://jocse.org/ especially for undergraduate research, though I don't know what "top" means here

**Reply by bhazelton** [reply]
+1 for JOSS

**Comment by namurphy** [comment]
- What communities exist akin to stackoverflow but for research software engineering?
   - US-RSE
   - Better Scientific Software

**Reply by erjank** [reply]
Software Carpentry really does have a lot of this community:

- https://carpentries.org/community-lessons/
- https://carpentries-incubator.github.io/hpc-intro/ 
- https://carpentries-incubator.github.io/python-intermediate-development/

**Reply by erjank** [reply]
Small followup discussion about how funded projects can ask for supplemental funding to support Carpentries efforts, and maybe SWC should apply for a Center of Excellence directly to support the CSSI communities at large

**Comment by namurphy** [comment]
- Idea of a hackathon, would that work well for a specialized project? 
   - Goal of Plasma Hack Week was broad, to have an opportunity for anyone to learn how to contribute to an open source project.  Some chance near end of the week to contribute to the package.
- Burnout is endemic, both among developers and in the broader community of potential contributors
- Projects more likely to be successful when the original developer gets someone else excited enough to contribute

**Comment by namurphy** [comment]
- How to protect intellectual property?
   - Different open source licenses have different clauses (e.g., copyleft)
   - Mixed models where some parts are open and some are for sale
   - Consultant model...people can release code as open source, and then make money from being a consultant

**Comment by gregtucker** [comment]
(The following are notes from one of the four tables)

Examples of motivation for community building:

- Form a professional (national or perhaps international level) community of Research Software Engineers
- Encourage users of a particular research software/CI project to become contributing developers
- Encourage other user-developers (or Research Software Engineers) to adopt a code that the creator no longer wants to devote substantial time to
- Coordinate and get better acquainted with/among a grass-roots user (and contributor?) community to a particular software
- Identify community members for a newly emerging project

Other notes:
- Idea of "time sharing a Research Software Engineer" (RSE): RSE's as a community resource who can "adopt" codes developed by researchers
- Definition of community can vary: e.g., domain science, a particular code, a particular general type or collection of codes, etc.
- Grants might last 2, 3, or 5 years, but individual people can be a 50-year investment

Modes of communication:
- Video conferencing is a game-changer that allows remote meeting, training sessions, etc.
- Jupyter notebooks and Hubs are similarly powerful in teaching and training
- Various other comm methods used around the table: email, videoconferencing (e.g., monthly "Tips and Tricks" video sessions; 2-hour training workshops), slack, discourse, google groups
